src/components/utils/src/timer.cc

```diff
@@ -64,6 +64,8 @@ timer::Timer::~Timer() {
   StopDelegate();
   single_shot_ = true;
 
+  thread_->set_delegate(NULL);
```

Comment:
@vlantonov I think these calls are redundant, but doublecheck `ThreadDelegate`'s destructor if you will remove them.